src/tools/annotation/bidirectionalTool/moveHandle/movePerpendicularLine/movePerpendicularLine.js

```diff
@@ -66,10 +64,10 @@ export default function movePerpendicularLine(
   );
 
   // Change the position of the perpendicular line handles
-  perpendicularStart.x = newLine.start.x;
-  perpendicularStart.y = newLine.start.y;
-  perpendicularEnd.x = newLine.end.x;
-  perpendicularEnd.y = newLine.end.y;
+  measurementData.handles.perpendicularStart.x = newLine.start.x;
+  measurementData.handles.perpendicularStart.y = newLine.start.y;
+  measurementData.handles.perpendicularEnd.x = newLine.end.x;
+  measurementData.handles.perpendicularEnd.y = newLine.end.y;
```

Comment:
This is still updating by reference ;)

Reply:
As per method description in JSDoc `Move the long line updating the perpendicular line handles position.`, I think it should update by reference.

How do you suggest to improve that implementation?
Because if I move to the parent function, it will still update by reference.

Reply:
I think I would expect any changes by reference to happen in the tool. Basically, wherever we "fetch" the reference, we should update it by reference (if we have to), in the same method. Isolating those kinds of changes to the tool itself feels logical.

I'm not opposed to alternative implementations -- feel free to discuss pros/cons of other approaches you might want to take.

Reply:
I just don't want to touch/refactor the upper levels of the tool.
Otherwise what should be a quick fix will become a huge refactor (which is already turning into).

Reply:
That's fine. We can table this for a different day :+1: